elements/campaign-display/components/logo.js

```diff
@@ -1,15 +1,43 @@
-import React, { PropTypes } from 'react';
-
-export default function Logo (props) {
-  let crop = props.crop || 'original';
-  let hasUrl = !!props.clickthrough_url;
-  let image = <div data-type="image" data-image-id={props.image_id} data-crop={crop}></div>;
-  let link = <a href={props.clickthrough_url}>{image}</a>;
-  return hasUrl ? link : image;
+import React, { PropTypes, Component } from 'react';
+
+let doPicturefill = function (component) {
+  window.picturefill(component.refs.image);
+};
+
+export default class Logo extends Component {
+
+  componentDidMount () {
+    // check if window.picturefill is available at mount time, otherwise wait
+    //  until the document is loaded, and hopefully image.js is loaded, and try
+    //  window.picturefill again
+    if (typeof window.picturefill === 'function') {
+      doPicturefill(this);
+    }
+    else {
+      window.addEventListener('load', doPicturefill.bind(null, this));
+    }
+  }
+
+  render () {
+    let crop = this.props.crop || 'original';
+    let hasUrl = !!this.props.clickthrough_url;
+    let image = (
```

Comment:
Agreed, let's get this out since we're close and work on refactoring it later. I'll take note of this change.